### PR TITLE
feat: adding support for karpenter blockdevicemapping

### PR DESCRIPTION
**Pull request by paulmowat**
*Description of changes:*

Adding in support to pass blockDeviceMappings through to karpenter as per https://karpenter.sh/v0.30/concepts/node-templates/#specblockdevicemappings

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/karpenter.test.ts [hunk 1]
```diff
@@ -114,42 +114,81 @@ describe('Unit tests for Karpenter addon', () => {
...
+    
+        const stack = await blueprint
+          .version("auto")
+          .account("123567891")
+          .region("us-west-1")
+          .addOns(
+            new blueprints.KarpenterAddOn({
+              version: 'v0.29.2',
+              values: {
+                blockDeviceMappings: [{
```
Comment (pennywisdom) [inline test/karpenter.test.ts:169]: I think this incorrectly setting the helm chart values instead of the addon blockDeviceMappings property in the test.

Comment (paulmowat) [reply test/karpenter.test.ts]: I've fixed this now.

File: lib/addons/karpenter/index.ts [hunk 2]
```diff
@@ -13,7 +13,26 @@ import * as sqs from 'aws-cdk-lib/aws-sqs';
...
+import { EbsDeviceVolumeType } from 'aws-cdk-lib/aws-ec2';
 
+interface BlockDeviceMapping {
+    deviceName?: string;
+    virtualName?: string;
+    ebs?: Ebs;
+    noDevice?: string;
+}
+  
+interface Ebs {
```
Comment (shapirov103) [inline lib/addons/karpenter/index.ts:25]: Ebs is a bit too bold for this interface  as it appears to represent ElasticBlockStorage as a service. How about either "EbsVolumeSettings` | `EbsVolumeMappings` or just `EbsMappings`?

File: lib/addons/karpenter/index.ts [hunk 3]
```diff
@@ -13,7 +13,26 @@ import * as sqs from 'aws-cdk-lib/aws-sqs';
 import { Rule } from 'aws-cdk-lib/aws-events';
 import { SqsQueue } from 'aws-cdk-lib/aws-events-targets';
 import { Cluster } from 'aws-cdk-lib/aws-eks';
+import { EbsDeviceVolumeType } from 'aws-cdk-lib/aws-ec2';
 
+interface BlockDeviceMapping {
```
Comment (shapirov103) [inline lib/addons/karpenter/index.ts:18]: These should be exported along with the `KarpenterAddOnProps` (not part of the PR atm) as customers will have issue consuming these otherwise.  Let's add `export` keyword to all public data types.

**Review by shapirov103 — COMMENTED** [review]
@paulmowat thank you for this contribution. Please see a couple relatively minor comments.

File: lib/addons/karpenter/index.ts (hunk 2 above)
Comment (paulmowat) [reply lib/addons/karpenter/index.ts]: thanks for the feedback. I've renamed to EBSVolumeMapping

File: lib/addons/karpenter/index.ts (hunk 3 above)
Comment (paulmowat) [reply lib/addons/karpenter/index.ts]: thanks for the feedback. I've exported all the interfaces (including KarpenterAddOnProps) and now use these as part of the unit test.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Comment by shapirov103** [comment]
Validated with unit tests, e2e will be used with the upgraded versions of helm/cdk in the 1.13 release prep.
